File: TsinghuaNetUWP.Helper/CredentialHelper.vb
Imports Windows.Security.Credentials

Public Module CredentialHelper
    Private Const CredentialResource As String = "TsinghuaNetUWP"
    Private vault As New PasswordVault

    Public Function GetCredential(username As String) As String
        For Each cre In From c In vault.RetrieveAll()
                        Where c.Resource = CredentialResource AndAlso c.UserName = username
            cre.RetrievePassword()
            Return cre.Password
        Next
        Return Nothing
    End Function

    Public Sub SaveCredential(username As String, password As String)
        vault.Add(New PasswordCredential(CredentialResource, username, password))
    End Sub

    Public Sub RemoveCredential(username As String)
        For Each cre In From c In vault.RetrieveAll()
                        Where c.Resource = CredentialResource AndAlso c.UserName = username
            vault.Remove(cre)
        Next
    End Sub
End Module

File: TsinghuaNet.Uno/TsinghuaNet.Uno.Helpers.Shared/CredentialHelper.cs
using System.Linq;
using Windows.Security.Credentials;

namespace TsinghuaNet.Uno.Helpers
{
    public static class CredentialHelper
    {
        private const string CredentialResource = "TsinghuaNetUWP";
        private static readonly PasswordVault vault = new PasswordVault();

        public static string GetCredential(string username)
        {
            foreach (var cre in from c in vault.RetrieveAll()
                                where c.Resource == CredentialResource && c.UserName == username
                                select c)
            {
                cre.RetrievePassword();
                return cre.Password;
            }
            return null;
        }

        public static void SaveCredential(string username, string password)
        {
            vault.Add(new PasswordCredential(CredentialResource, username, password));
        }

        public static void RemoveCredential(string username)
        {
            foreach (var cre in from c in vault.RetrieveAll()
                                where c.Resource == CredentialResource && c.UserName == username
                                select c)
            {
                vault.Remove(cre);
            }
        }
    }
}
